File: Componentes/FlexCompress 6.94 D6-D11/Demos/Delphi/Basic/Dest/uMain.pas
unit uMain;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls, Buttons, FlexCompress, uProgress;

type
  TfmMain = class(TForm)
    Archiver: TFlexCompress;
    bnStart: TBitBtn;
    BitBtn2: TBitBtn;
    Label1: TLabel;
    procedure bnStartClick(Sender: TObject);
    procedure BitBtn2Click(Sender: TObject);
    procedure ArchiverFileProgress(Sender: TObject; FileName: String;
      Progress: Double; Operation: TFXCProcessOperation;
      ProgressPhase: TFXCProgressPhase; var Cancel: Boolean);
    procedure ArchiverOverallProgress(Sender: TObject; Progress: Double;
      Operation: TFXCProcessOperation; ProgressPhase: TFXCProgressPhase;
      var Cancel: Boolean);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  fmMain: TfmMain;

implementation

{$R *.DFM}

procedure TfmMain.bnStartClick(Sender: TObject);
begin
 with Archiver do
    begin
        FileName := 'Archive\test.fxc';

        // Create a new archive file
        OpenArchive(fmCreate);

        // Set path to folder with some text files to BaseDir
        BaseDir := 'Source';

        // Add all files and directories from 'C:\SOURCE_FOLDER' to the archive
        AddFiles('*.*');

        // Set path to destination folder
        BaseDir := 'Dest';

        // Extract all files and directories from the archive to 'C:\DESTINATION_FOLDER'
        ExtractFiles('*.*');

        // Close the archive
        CloseArchive;
    end;
    ShowMessage('All files were added and extracted successfully.');
end;

procedure TfmMain.BitBtn2Click(Sender: TObject);
begin
 Close;
end;

procedure TfmMain.ArchiverFileProgress(Sender: TObject; FileName: String;
  Progress: Double; Operation: TFXCProcessOperation;
  ProgressPhase: TFXCProgressPhase; var Cancel: Boolean);
begin
 frmProgress.gFile.Progress := Round(Progress);
 frmProgress.lbFile.Caption := FileName;
 Cancel := frmProgress.bCancel;
 Application.ProcessMessages;
end;

procedure TfmMain.ArchiverOverallProgress(Sender: TObject;
  Progress: Double; Operation: TFXCProcessOperation;
  ProgressPhase: TFXCProgressPhase; var Cancel: Boolean);
begin
 if (ProgressPhase = ppStart) then
  frmProgress.Show
 else
 if (ProgressPhase = ppEnd) then
  frmProgress.Hide;

 frmProgress.gOverall.Progress := Round(Progress);
 Cancel := frmProgress.bCancel;
 Application.ProcessMessages;
end;

end.

File: Componentes/ZipForge 6.94 for D6-D11/Demos/C++ Builder 2010/TransactionSystem/uMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "uMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "ZipForge"
#pragma resource "*.dfm"
TfmMain *fmMain;
//---------------------------------------------------------------------------
__fastcall TfmMain::TfmMain(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfmMain::bnStartClick(TObject *Sender)
{
 Archiver->FileName = "Archive\\test.zip";

 // Create a new archive file
 Archiver->OpenArchive(fmCreate);

 // Start a transaction
 Archiver->BeginUpdate();

 // Set path to folder with some HTML files to BaseDir
 Archiver->BaseDir = "Source";

 // Add all files from Source folder to the archive
 try
 {
  Archiver->AddFiles("*.*");
 }
 catch(...)
 {
  // If errors occurs rollback transaction. All modifications will be cancelled.
  Archiver->CancelUpdate();

  // Close archive and exit current procedure
  Archiver->CloseArchive();
  ShowMessage("Error adding all files");
  return;
 }

 // Set path to folder with some HTML files to BaseDir
 Archiver->BaseDir = "Source1\\";

 // Add all HTML files from Source1 folder to the archive
 try
 {
  Archiver->AddFiles("*.htm*");
 }
 catch(...)
 {
   // If errors occurs rollback transaction. All modifications will be cancelled.
   Archiver->CancelUpdate();

   // Close archive and exit current procedure
   Archiver->CloseArchive();
   ShowMessage("Error adding html files");
   return;
 }

 // Commit a transaction. All modifications will be saved.
 Archiver->EndUpdate();

 // Set path to destination folder
 Archiver->BaseDir = "Dest";

 // Extract all files
 Archiver->ExtractFiles("*.*");

 // Close the archive
 Archiver->CloseArchive();

 ShowMessage("All files were added and extracted successfully.");
}
//---------------------------------------------------------------------------
void __fastcall TfmMain::BitBtn2Click(TObject *Sender)
{
 Close();
}
//---------------------------------------------------------------------------
